src/source/JavaGenerator.scala

```diff
@@ -140,7 +140,10 @@ class JavaGenerator(spec: Spec) extends Generator(spec) {
       writeDoc(w, doc)
 
       javaAnnotationHeader.foreach(w.wl)
-      w.w(s"${javaClassAccessModifierString}abstract class $javaClass$typeParamList").braced {
+      val isAbstractClass = (spec.javaGenerateInterfaces == false) || i.ext.cpp || i.methods.exists(_.static) || i.consts.size > 0
```

Comment:
This could use a comment explaining the reason for each of the clauses, and why an interface can't support them.

Also, a Java interface can have static final member variables, so you should be able to generate an interface in that case.  In fact, static and final are implied for member variables in interfaces, so the generator should probably leave them off to avoid warnings.